webapp/src/ts/services/browser-detector.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import * as Bowser from 'bowser';
+import { Store } from '@ngrx/store';
+import { Selectors } from '@mm-selectors/index';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BrowserDetectorService {
+  private _parser: Bowser.Parser.Parser;
+  private androidAppVersion: VersionNumber | undefined;
+
+  public constructor(private store: Store) {
+    this.store.select(Selectors.getAndroidAppVersion).subscribe(androidAppVersion => {
+      this.androidAppVersion = androidAppVersion;
+    });
+  }
+
+  private get parser() {
+    // lazy initialization to allow unit tests to use a different user agent
+    if (typeof this._parser === 'undefined') {
+      this._parser = Bowser.getParser(window.navigator.userAgent);
+    }
+
+    return this._parser;
+  }
+
+  public detect() {
+    return this.parser.getBrowser();
+  }
+
+  public isUsingSupportedBrowser() {
+    return this.parser.satisfies({
+      chrome: '>=53',
+      firefox: '>=98', // latest at the time
+    });
+  }
+
+  public isUsingChtAndroid() {
+    return typeof this.androidAppVersion !== 'undefined';
+  }
+
+  public isUsingChtAndroidV1() {
```

Comment:
This seems to not be used anywhere. It it still needed?

Reply:
This code block is not used except in tests and I would rather keep it as the logic has been copied to the login page `script.js` and it's being used to show a specific message for users using the CHT with `cht-android` below v1

Removing it here in webapp won't break anything but the associated unit tests helped me reach this solution and can serve as an explanation

Reply:
> serve as an explanation

I'm generally not in favor of leaving testing code lying around. I'm not entirely sure what explanation you're referring to.
Since the two pieces of code are completely independent, we shouldn't rely on a unit test testing unused code to prove that code somewhere  else works as expected.

Reply:
It makes sense so I've followed your advice and got rid of the unused code